### ISSUE TITLE
plotIndiv with PLSDA and sPLSDA color issue when the number of groups is large (> 3)

### ISSUE THREAD

**Issue by mixOmicsTeam** [mixOmicsTeam/mixOmics#159]
Edit: the error msg was a bit confusing for me to work it out (especially because a plotIndiv.pca() would not complain if the group information included NA values!).

------------

🐞  
Data are here: https://cloudstor.aarnet.edu.au/plus/s/7mgKK5MHDcjyMun
Could we add a check / update the message error for all PLS-DA, sPLS-DA and associated functions (e.g. tune, perf)?

------------
🔍 

packageVersion('mixOmics') [1] ‘6.18.0’

```
library(mixOmics)
load('nmt_data_processed.RData')
Y = cell_metadata$lineage
plsda.rna.result = plsda(X = data$rna, Y = Y, ncomp = 3)
plotIndiv(plsda.rna.result)
Error: Unknown colour name: FALSE
```

(sorry, cant do reprex because I m using the Rcloud)

------------

🤔  **Expected behavior:**
```
plotIndiv(plsda.rna.result, group = cell_metadata$lineage, pch = 16) # works
plotIndiv(plsda.rna.result) # does not work
plotIndiv(plsda.rna.result, col.per.group = c(1:8)) # does not work
```

**Comment by aljabadi** [comment]
@mixOmicsTeam this issue must be fixed as of `6.18.1`

**Comment by mixOmicsTeam** [comment]
I am closing this issue. I had provided an update that the problem came from the fact that the Y outcome had missing values in the DA models and that posed a problem for the plots (gastrulation data, lineage info).
